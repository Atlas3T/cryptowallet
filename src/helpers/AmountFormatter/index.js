/**
 * @TODO James Konrad
 * I've mocked some exchange rates below for the amount() function to use
 * Would make sense to put them somewhere in the Vuex store
 */

import numeral from 'numeral';

const exchangeRates = {
  BTC: [
    {
      currency: 'USD',
      rate: 6431.14,
    },
    {
      currency: 'GBP',
      rate: 4977.06,
    },
  ],
  ETH: [
    {
      currency: 'USD',
      rate: 201.76,
    },
    {
      currency: 'GBP',
      rate: 156.14,
    },
  ],
  ATL: [
    {
      currency: 'USD',
      rate: 22222,
    },
    {
      currency: 'GBP',
      rate: 9999,
    },
  ],
};

export default class AmountFormatter {
  constructor(options) {
    this.amount = options.amount;
    this.coin = options.coin;
    this.currency = options.currency;
    this.format = options.format;
    this.withCurrencySymbol = options.withCurrencySymbol;
    this.prependPlusOrMinus = options.prependPlusOrMinus;
  }

  /**
   * Formats a Number and returns a String
   * @return {String}
   */
  getFormatted() {
    let { amount } = this;
    if (this.coin && this.currency) {
      amount = this.coinToCurrency(this.amount, this.coin, this.currency);
    }

    let formatted = `${numeral(Math.abs(amount)).format(this.format)}`;
    if (this.currency) formatted = `${this.currency.symbol}${formatted}`;

    if (this.prependPlusOrMinus) {
      if (amount < 0) return `- ${formatted}`;
      if (amount > 0) return `+ ${formatted}`;
    }

    return formatted;
  }

  /**
   * Converts a coin into a currency
   * @return {Number}
   */
  coinToCurrency() {
    const { rate } = exchangeRates[this.coin].find(item => item.currency === this.currency.code);
    if (rate) return this.amount * rate;
    return this.amount;
  }
}
