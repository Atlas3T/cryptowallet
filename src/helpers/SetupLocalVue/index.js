import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
