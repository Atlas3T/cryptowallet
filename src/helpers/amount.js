/**
 * @TODO James Konrad
 * mocked data below
 */

import numeral from 'numeral';

const currencySymbols = {
  USD: '$',
  GPB: 'Â£',
};

const exchangeRates = {
  BTC: [
    {
      currency: 'USD',
      rate: 6431.14,
    },
    {
      currency: 'GPB',
      rate: 4977.06,
    },
  ],
  ETH: [
    {
      currency: 'USD',
      rate: 201.76,
    },
    {
      currency: 'GPB',
      rate: 156.14,
    },
  ],
  ATL: [
    {
      currency: 'USD',
      rate: 22222,
    },
    {
      currency: 'GPB',
      rate: 9999,
    },
  ],
};

export default class Amount {
  constructor(options) {
    this.amount = options.amount;
    this.coin = options.coin;
    this.currency = options.currency;
    this.format = options.format;
    this.withCurrencySymbol = options.withCurrencySymbol;
    this.prependPlusOrMinus = options.prependPlusOrMinus;
  }

  /**
   * Formats a Number and returns a String
   * @return {String}
   */
  get formatted() {
    let { amount } = this;
    if (this.coin && this.currency) {
      amount = this.coinToCurrency(this.amount, this.coin, this.currency);
    }

    let formatted = `${numeral(Math.abs(amount)).format(this.format)}`;
    if (this.currency) formatted = `${currencySymbols[this.currency]}${formatted}`;

    if (this.prependPlusOrMinus) {
      if (amount < 0) return `&#45; ${formatted}`;
      if (amount > 0) return `&#43; ${formatted}`;
    }

    return formatted;
  }

  /**
   * Converts a coin into a currency
   * @return {Number}
   */
  coinToCurrency() {
    const { rate } = exchangeRates[this.coin].find(item => item.currency === this.currency);
    return this.amount * rate;
  }
}
