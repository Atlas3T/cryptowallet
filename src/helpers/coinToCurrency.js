const exchangeRates = {
  btc: [
    {
      currency: 'USD',
      value: 6431.14,
    },
    {
      currency: 'GPB',
      value: 4977.06,
    },
  ],
  eth: [
    {
      currency: 'USD',
      value: 201.76,
    },
    {
      currency: 'GPB',
      value: 156.14,
    },
  ],
  atlas: [
    {
      currency: 'USD',
      value: 22222,
    },
    {
      currency: 'GPB',
      value: 9999,
    },
  ],
};

/**
 * Calcuates the value of a coin in a given currency
 * @param  {Number} amount
 * @param  {String} coin
 * @param  {String} currency
 * @return {Number}
 */

export default function coinToCurrency(amount, coin, currency) {
  const { value } = exchangeRates[coin].find(item => item.currency === currency);
  return amount * value;
}
