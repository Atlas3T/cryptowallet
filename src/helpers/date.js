import { date as dateUtil } from 'quasar';

/**
 * Returns formatted and translated date string
 * @param  {Number} timestamp
 * @param  {String} format
 * @param  {Object} vm Vue component instance
 * @return {String}
 */
export default function date(timestamp, format, vm) {
  return dateUtil.formatDate(timestamp, format, {
    dayNames: [
      'sunday',
      'monday',
      'tuesday',
      'wednesday',
      'thursday',
      'friday',
      'saturday',
    ].map(item => vm.$t(item)),
    monthNames: [
      'january',
      'february',
      'march',
      'april',
      'may',
      'june',
      'july',
      'august',
      'september',
      'october',
      'november',
      'december',
    ].map(item => vm.$t(item)),
  });
}
