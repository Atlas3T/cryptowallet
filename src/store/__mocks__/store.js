import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
