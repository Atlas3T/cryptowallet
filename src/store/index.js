/*eslint-disable*/

import Vue from 'vue';
import Vuex from 'vuex';
import VuexPersistence from 'vuex-persist';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from 'vuex-orm-lokijs';

// import entities.
import Account from './wallet/entities/account';
import Address from './wallet/entities/address';
import Tx from './wallet/entities/tx';
import Utxo from './wallet/entities/utxo';
import Wallet from './wallet/entities/wallet';
import Coin from './wallet/entities/coin';
import KeyPair from './wallet/entities/keyPair';
import Prices from './prices';
import LatestPrice from './latestPrice';

// import modules.
import payments from './payments';
import search from './search';
import settings from './settings';
import setup from './setup';
import contacts from './contacts';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
];

window.Tx = Tx;
window.Utxo = Utxo;
window.Address = Address;
window.Wallet = Wallet;
window.LatestPrice = LatestPrice;
window.Prices = Prices;
window.Account = Account;
window.Coin = Coin;

// Setup persistant storage.
const vuexLocal = new VuexPersistence({
  supportCircular: true,
  storage: window.localStorage,
  modules: [settings],
});


const options = {
  env: 'browser',
};

function hydrationCompletedCallback() {
  setTimeout(() => {
    store.dispatch('settings/setLoading', false);
  }, 1000);
}

VuexORM.use(VuexORMLoki, { database, options, hydrationCompletedCallback });


/**
 * Create CryptoWallet Vuex store obj.
 * @type {Store<any>}
 */
const store = new Vuex.Store({
  modules: {
    payments,
    search,
    settings,
    setup,
    contacts,
  },
  plugins: [VuexORM.install(database)],
});

Vue.prototype.activeWallets = {};

// @todo remove line below (used for development purposes only)
window.store = store;

export default store;
