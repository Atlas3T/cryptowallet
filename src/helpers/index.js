import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
