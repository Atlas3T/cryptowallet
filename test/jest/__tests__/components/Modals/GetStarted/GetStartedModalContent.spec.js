import { mount } from '@vue/test-utils';
import GetStartedModalContent from '@/components/Modals/GetStarted/GetStartedModalContent';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';

describe('GetStartedModalContent component', () => {
  let wrapper;
  let router;
  let store;
  let storeMocks;

  const delay = 50;

  const defaultProps = {};

  function wrapperInit(options) {
    return mount(GetStartedModalContent, options);
  }

  function storeInit(custom, propsData, parentComponent = null) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: '/setup/0' });
    wrapper = wrapperInit({
      i18n,
      router,
      localVue,
      propsData,
      parentComponent,
      store: storeMocks.store,
    });
    store = wrapper.vm.$store;
  }

  beforeEach(() => { storeInit({}, defaultProps); });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('updates state when network status changes', async (done) => {
    window.dispatchEvent(new Event('offline'));

    setTimeout(() => {
      expect(wrapper.vm.online).toBe(false);
      window.dispatchEvent(new Event('online'));

      setTimeout(() => {
        expect(wrapper.vm.online).toBe(true);
        done();
      }, delay);
    }, delay);
  });

  it('switches slides when "NEXT" button is clicked', () => {
    const current = wrapper.vm.slide;
    wrapper.find('.splash-btn').trigger('click');
    expect(wrapper.vm.slide !== current).toBe(true);
  });

  it('enables the "ACTIVATE YOUR WALLET" button when network goes offline; calls done() method when clicked', () => {
    wrapper.vm.slide = 'security';
    expect(wrapper.contains('.splash-btn[disabled=disabled]')).toBe(true);
    window.dispatchEvent(new Event('offline'));
    expect(wrapper.contains('.splash-btn[disabled=disabled]')).toBe(false);
    expect(wrapper.contains('.splash-btn')).toBe(true);

    const doneMock = jest.fn();
    wrapper.setMethods({ done: doneMock });
    wrapper.find('.splash-btn').trigger('click');
    expect(doneMock).toHaveBeenCalled();
  });

  describe('done() method', () => {
    it('dispatches setGetStartedModalOpened with false as payload', () => {
      wrapper.vm.done();
      expect(storeMocks.actions.setGetStartedModalOpened.mock.calls[0][1]).toBe(false);
    });

    it('dispatches setAccountType with \'new\' as payload', () => {
      wrapper.vm.done();
      expect(storeMocks.actions.setAccountType.mock.calls[1][1]).toBe('new');
    });

    it('navigates to /setup/2 path', () => {
      wrapper.vm.done();
      expect(store.state.route.path).toBe('/setup/2');
    });
  });
});
