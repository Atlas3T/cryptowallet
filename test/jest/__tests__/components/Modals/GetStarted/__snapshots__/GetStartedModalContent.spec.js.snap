// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStartedModalContent component renders and matches snapshot 1`] = `
<div
  class="modal-layout-wrapper full no-padding"
>
  <div
    class="q-carousel q-panel-parent q-carousel-full q-carousel--navigation"
  >
    <div
      class="q-carousel__slides-container"
    >
      <div
        class="q-panel scroll"
        role="tabpanel"
      >
        <div
          class="q-carousel__slide full-slide"
        >
          <div
            class="slide-header"
          >
            <h1
              class="setup"
            >
              
          Coins & Tokens
        
            </h1>
          </div>
           
          <div
            class="slide-wrapper"
          >
            <div
              class="slide-illustration"
            >
              <i
                aria-hidden="true"
                class="material-icons q-icon text-white temp-ill"
                style="font-size: 10rem;"
              >
                monetization_on
              </i>
            </div>
             
            <p>
              
          Digital currencies that are secured using cryptography and stored on a distributed ledger, blockchain or DAG. It's digital money. Used by millions world-wide.
        
            </p>
          </div>
           
          <div
            class="absolute-bottom"
          >
            <div
              class="next-button-wrapper"
            >
              <button
                class="q-btn inline q-btn-item non-selectable splash-btn q-btn--standard q-btn--rectangle bg-yellow text-blueish q-focusable q-hoverable"
                tabindex="0"
                type="button"
              >
                <div
                  class="q-focus-helper"
                  tabindex="-1"
                />
                <div
                  class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
                >
                  <div>
                    Next
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="q-carousel__control q-carousel__navigation no-wrap absolute flex scroll-x q-carousel__navigation--buttons text-white"
    >
      <div
        class="q-carousel__navigation-inner flex no-wrap justify-center"
      >
        <button
          class="q-btn inline q-btn-item non-selectable q-carousel__navigation-icon q-btn--flat q-btn--round q-focusable q-hoverable q-carousel__navigation-icon--active"
          style="font-size: 10px;"
          tabindex="0"
          type="button"
        >
          <div
            class="q-focus-helper"
            tabindex="-1"
          />
          <div
            class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
          >
            <i
              aria-hidden="true"
              class="material-icons q-icon"
            >
              lens
            </i>
          </div>
        </button>
        <button
          class="q-btn inline q-btn-item non-selectable q-carousel__navigation-icon q-btn--flat q-btn--round q-focusable q-hoverable"
          style="font-size: 10px;"
          tabindex="0"
          type="button"
        >
          <div
            class="q-focus-helper"
            tabindex="-1"
          />
          <div
            class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
          >
            <i
              aria-hidden="true"
              class="material-icons q-icon"
            >
              lens
            </i>
          </div>
        </button>
        <button
          class="q-btn inline q-btn-item non-selectable q-carousel__navigation-icon q-btn--flat q-btn--round q-focusable q-hoverable"
          style="font-size: 10px;"
          tabindex="0"
          type="button"
        >
          <div
            class="q-focus-helper"
            tabindex="-1"
          />
          <div
            class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
          >
            <i
              aria-hidden="true"
              class="material-icons q-icon"
            >
              lens
            </i>
          </div>
        </button>
        <button
          class="q-btn inline q-btn-item non-selectable q-carousel__navigation-icon q-btn--flat q-btn--round q-focusable q-hoverable"
          style="font-size: 10px;"
          tabindex="0"
          type="button"
        >
          <div
            class="q-focus-helper"
            tabindex="-1"
          />
          <div
            class="q-btn__content text-center col items-center q-anchor--skip justify-center row"
          >
            <i
              aria-hidden="true"
              class="material-icons q-icon"
            >
              lens
            </i>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
